cpp/include/raft/linalg/map.cuh

```diff
@@ -37,17 +39,63 @@ namespace linalg {
  * @param in the input array
  * @param args additional input arrays
  */
+template <typename InType,
+          typename MapOp,
+          typename IdxType = std::uint32_t,
+          int TPB          = 256,
+          typename OutType = InType,
+          typename... Args>
+void map_k(
+  OutType* out, IdxType len, MapOp map, cudaStream_t stream, const InType* in, Args... args)
+{
+  detail::mapImpl<InType, OutType, IdxType, MapOp, TPB, Args...>(
+    out, len, map, stream, in, args...);
+}
+
+/**
+ * @defgroup map Mapping ops
+ * @{
+ */
 
+/**
+ * @brief CUDA version of map
+ * @tparam InType data-type for math operation of type raft::device_mdspan
+ * @tparam MapOp the device-lambda performing the actual operation
+ * @tparam TPB threads-per-block in the final kernel launched
+ * @tparam OutType data-type of result of type raft::device_mdspan
+ * @tparam Args additional parameters
+ * @param handle raft::handle_t
+ * @param out the output of the map operation of type raft::device_mdspan
+ * @param map the device-lambda
+ * @param in the input of type raft::device_mdspan
+ * @param args additional input arrays
+ */
 template <typename InType,
           typename MapOp,
-          int TPB = 256,
+          int TPB          = 256,
+          typename OutType = InType,
           typename... Args,
-          typename OutType = InType>
-void map(OutType* out, size_t len, MapOp map, cudaStream_t stream, const InType* in, Args... args)
+          typename = enable_if_device_mdspan<InType, OutType>>
+void map(const raft::handle_t& handle, OutType out, MapOp map, const InType in, Args... args)
```

Comment:
```suggestion
void map(const raft::handle_t& handle, OutType out, MapOp map, InType in, Args... args)
```
Please see notes above about using `mdspan<const T, ...>` instead of `const mdspan<T, ...>`; thanks!